### ISSUE TITLE
url error when connect with query

### ISSUE THREAD

**Issue by czcz1024** [DyKnow/SignalR-ObjC#194]
when connect with querystring
url like this
http://server/signalr/negotiate&a=1&b=2
it should be
http://server/signalr/negotiate?a=1&b=2

**Comment by backofthecup** [comment]
@czcz1024 
Any update on this? I'm having the same issue with 2.0.0.beta1.

**Comment by joeldart** [comment]
Responding to comment in #198 I don't have any update at this point but where's the error in the console log? It SRLogs the url before doing the connect/reconnect. What expected/actual url you seeing?

Right now I'm concentrating on nailing down more of the flow control stuff and retry events by adding the unit tests/various fixes.

**Comment by backofthecup** [comment]
Here it is:
````
NSLocalizedDescription=Request failed: not found (404)}
2015-08-04 12:37:43.707 PatientMobile[3326:83343] -[CCSignalRService SRConnection:didReceiveError:] line 218 $ SignalR Error connecting to <authenticatedUserHub>: Error Domain=com.alamofire.error.serialization.response Code=-1011 "Request failed: not found (404)" UserInfo=0x7faafaf9ac10 {NSUnderlyingError=0x7faafac492c0 "Request failed: unacceptable content-type: text/html", com.alamofire.serialization.response.error.response=<NSHTTPURLResponse: 0x7faafaebeb60> { URL: https://dev.carecamhealthsystems.com/API/login.aspx?ReturnUrl=%2fapi%2fsignalr%2fsignalr%2fnegotiate%3fapiKey%3dF8F90649-8EFF-447B-8200-A837D9406369%26authToken%3dMtwIjwOVAQwSgB8c3YbAGfhesXDFf9EG%25252F%25252FVRykqjqoGBEdo3puOBviUIgAsK0o4p%25252BCMSU11DVpxUI%25252FiKHGt9uA%25253D%25253D%26clientProtocol%3d1.3.0.0%26connectionData%3d%255B%257B%2522Name%2522%253A%2522authenticateduserhub%2522%257D%255D&apiKey=F8F90649-8EFF-447B-8200-A837D9406369&authToken=MtwIjwOVAQwSgB8c3YbAGfhesXDFf9EG%252F%252FVRykqjqoGBEdo3puOBviUIgAsK0o4p%252BCMSU11DVpxUI%252FiKHGt9uA%253D%253D&clientProtocol=1.3.0.0&connectionData=%5B%7B%22Name%22%3A%22authenticateduserhub%22%7D%5D } { status code: 404, headers {
    "Access-Control-Allow-Headers" = "Authorization,Origin, X-Requested-With, Content-Type, Accept";
    "Access-Control-Allow-Methods" = "GET,PUT,POST,DELETE,OPTIONS";
    "Cache-Control" = private;
    "Content-Length" = 3212;
    "Content-Type" = "text/html; charset=utf-8";
    Date = "Tue, 04 Aug 2015 17:37:44 GMT";
    Server = "Microsoft-IIS/8.0";
    "X-AspNet-Version" = "4.0.30319";

````

**Comment by joeldart** [comment]
Looks like forms auth error to me, also not something I have experience with. You could probably use [connection setHeaders: NSDictionary*] to pass up the forms auth token

**Comment by backofthecup** [comment]
@joeldart 

First of all, the negotiate URL is not logged. I modified the code the log it.

Here is the negotiate URL for <b>2.0.0.alpha1

```
.....NEGOTIATE REQUEST URL.... https://<somehost>/signalr/negotiate?authToken=MtwIjwOVAQwSgB8c3YbAGdgRkqXIxU7xorNXngwwojBy4ulUL7PJxcsDt4FY%2BuFVxtqUUHHpoA%2FrqHdAP81YMg%3D%3D
&apiKey=xxxxx
&clientProtocol=1.3.0.0
&connectionData=%5B%7B%22Name%22%3A%22authenticateduserhub%22%7D%5D

````

Here is the negotiate URL for <b>2.0.0.beta1
````
.....NEGOTIATE REQUEST URL.... https://<somehost>/signalr/negotiate?apiKey=xxxxx
&authToken=MtwIjwOVAQwSgB8c3YbAGdgRkqXIxU7xorNXngwwojBy4ulUL7PJxcsDt4FY%252BuFVxtqUUHHpoA%252FrqHdAP81YMg%253D%253D
&clientProtocol=1.3.0.0
&connectionData=%5B%7B%22Name%22%3A%22authenticateduserhub%22%7D%5D
````

Notice the <b>authTokens</b> are different. It looks like in version 2.0.0.alpha1 the queryString passed in is expected to be URL encoded, whereas, in version 2.0.0.beta1 the queryString is being URL encoded. Thus, my authToken is being URL encoded twice. 

The problem is no one tested the case where a queryString contained a string that needed to be URL encoded. Please add that to your test cases.

**Comment by joeldart** [comment]
@backofthecup I think I got it. to clarify, we are now urlencoding the query params where we were not before, so you can pass in the dictionary without doing the encoding yourself. Is that correct? That change was a breaking api change for your app as the prior api required url encoding be done in the app itself.

**Comment by backofthecup** [comment]
Yes. That is exactly it. Once I passed in my queryString dictionary unencoded, it worked. I recognized this because, we also use AFNetworking, which is encoding those parameters when using a NSDictionary. That's why I was suspicious of that from the beginning.  My only suggestion would be to document that method argument explicitly saying the queryString dictionary should not be encoded. 

Problem is solved and things are back to normal.  In the end, I learned a lot about your API. So it was a great learning experience. Thanks for your help.
